PROJECT_HEURISTICS.md

```diff
@@ -0,0 +1,261 @@
+# Project heuristics
+
+Vscode and vim need to be able to detect the correct location of the `.compiler.log`, `bsc.exe` and `bsb` for diagnostics and formatting.
+Different project layouts result into different locations. The aim of this document is to formalize the different cases and specify where the editor tooling should search for `.compiler.log`, `bsc.exe` and `bsb`
+
+## Heuristics
+
+1. `.compiler.log` (diagnostics)
+- Starting from the current open file, walk the filesystem upwards, until `bsconfig.json` is located
+- `.compiler.log` is located at `path.join(foundDirectory, "lib", "bs", ".compiler.log")`
+
+2. platform specific `bsc.exe` (formatter)
+- Starting from the current open file, walk the filesystem upwards, until `./node_modules/bs-platform` is located
+- `bsc.exe` is located at `path.join(foundDirectory, "./node_modules/bs-platform", platform, "bsc.exe")`
+
```

Comment:
This is not exact since if you are not in the root project but in a dependency, this might be wrong

Reply:
I have a vague memory that there is something called `path.resolve`